src/Microsoft.Identity.Web/DownstreamWebApiSupport/MicrosoftIdentityAuthenticationMessageHandlerHttpClientBuilderExtensions.cs

```diff
@@ -32,13 +31,12 @@ public static class MicrosoftIdentityAuthenticationMessageHandlerHttpClientBuild
             }
 
             builder.Services.Configure<MicrosoftIdentityAuthenticationMessageHandlerOptions>(serviceName, configuration);
+            builder.Services.AddSingleton<IMicrosoftIdentityAuthenticationDelegatingHandlerFactory, MicrosoftIdentityAuthenticationDelegatingHandlerFactory>();

```

Comment:
You should use `TryAddSingleton` on `IMicrosoftIdentityAuthenticationDelegatingHandlerFactory` so that a user can configure his own implementation before and we will not override it.

As this is done in all extension methods maybe put this in a `AddMicrosoftIdentityAuthenticationHandlerCore` private method called by all other extension methods so that common configuration is shared.